nomad/rpc.go

```diff
@@ -140,6 +147,34 @@ func (ctx *RPCContext) ValidateCertificateForName(name string) error {
 	return fmt.Errorf("invalid certificate, %s not in %s", name, strings.Join(validNames, ","))
 }
 
+func (ctx *RPCContext) IsStatic() bool {
+	return ctx == nil
+}
+
+func (ctx *RPCContext) GetRemoteIP() (net.IP, error) {
+	var remoteAddr *net.TCPAddr
+	var ok bool
```

Comment:
I don't know how often `ctx` will be nil, but if it's a decent amount, is it worth moving these past that check?

Reply:
`ctx` is only ever nil when we're sending "static" RPCs. That is, RPCs that are always sent as (expensive) method calls rather than being sent over the wire. An example of that is the deployment watcher, which can use that for all its local RPCs because it only runs on the leader.

So that's probably a good chunk. Can't hurt to move it.